pkgs/development/libraries/qt-6/default.nix

```diff
@@ -167,10 +167,19 @@ let
       qtwebview = callPackage ./modules/qtwebview.nix { };
 
       wrapQtAppsHook = callPackage
-        ({ makeBinaryWrapper }: makeSetupHook
+        ({ makeBinaryWrapper, qtwayland }: makeSetupHook
           {
             name = "wrap-qt6-apps-hook";
             propagatedBuildInputs = [ makeBinaryWrapper ];
+            depsTargetTargetPropagated = lib.optionals (lib.meta.availableOn stdenv.targetPlatform qtwayland) [ qtwayland.dev ];
```

Comment:
Just checking: `.dev` is definitely correct here?

Reply:
I think qtbase is still necessary, so the plugins from it can be found.

Reply:
> I think qtbase is still necessary, so the plugins from it can be found.

Both of these launch
```
$ env -i HOME="$HOME" DISPLAY="$DISPLAY" WAYLAND_DISPLAY="$WAYLAND_DISPLAY" XDG_RUNTIME_DIR="$XDG_RUNTIME_DIR" QT_QPA_PLATFORM=xcb PATH="$PATH" QT_PLUGIN_PATH=/nix/store/kgz360i4bwc19c35ag7m35vn7f0910k2-qtwayland-6.8.0/lib/qt-6/plugins ./result/bin/.ppic-wrapped
$ env -i HOME="$HOME" DISPLAY="$DISPLAY" WAYLAND_DISPLAY="$WAYLAND_DISPLAY" XDG_RUNTIME_DIR="$XDG_RUNTIME_DIR" QT_QPA_PLATFORM=wayland PATH="$PATH" QT_PLUGIN_PATH=/nix/store/kgz360i4bwc19c35ag7m35vn7f0910k2-qtwayland-6.8.0/lib/qt-6/plugins ./result/bin/.ppic-wrapped
```

Reply:
I guess it's fine to just have it pulled from the application's dependencies.

Reply:
Propagated build inputs only propagate one level, right? I can conceive of some package that encapsulates its use of Qt so that the actual application doesn’t have an explicit `qtbase` dependency and would thus break.

Reply:
It would be good if we only propagated the plugins dir

Reply:
Well that isn't possible because the plugins depend on $out

Reply:
It seems fine just to propagate `qtbase.out` and `qtwayland.out` for now, right? I doubt the closure size from the header files matters much?

Reply:
I'm not concerned with closure size, instead I'm concerned with env pollution

Reply:
https://github.com/NixOS/nixpkgs/pull/352466